packages/vite/src/node/server/middlewares/proxy.ts

```diff
@@ -194,24 +200,26 @@ export function proxyMiddleware(
   }
 
   // Keep the named function. The name is visible in debug logs via `DEBUG=connect:dispatcher ...`
-  return function viteProxyMiddleware(req, res, next) {
+  return async function viteProxyMiddleware(req, res, next) {
```

Comment:
We need to wrap the whole handler with `try-catch` and call `next(error)` in the `catch`, so that `next` is called  when an error happens asynchronously.

Reply:
This means that errors in `bypass` will ignore the entire behavior of the `proxy`, both in the `proxyMiddleware` and during the http server event `upgrade`.